src/services/domain/cart.service.js

```diff
@@ -1,7 +1,138 @@
 const prisma = require('../db/prisma')
+const cron = require('node-cron')
 
 class CartService {
+  async releaseOldReservation() {
+    const oneWeekAgo = new Date()
+    oneWeekAgo.setDate(oneWeekAgo.getDate() - 7)
+    console.log('The task of removing the old armor is completed')
+
+    const reservedProducts = await prisma.cartItem.findMany({
+      where: {
+        createdAt: {
+          lt: oneWeekAgo,
+        },
+      },
+    })
+
+    for (const item of reservedProducts) {
+      await prisma.product.update({
+        where: {
+          id: item.productId,
+        },
+        data: {
+          quantity: {
+            increment: item.quantity,
+          },
+        },
+      })
+      await prisma.cartItem.delete({
+        where: { id: item.id },
+      })
+    }
+    console.log(
+      'The release of the goods and the quantity update have been completed',
+    )
+  }
+
+  scheduleReleaseOfOldReservations() {
+    cron.schedule(
+      '0 0 * * *',
+      async () => {
+        console.log(
+          '[CronJob]: The beginning of the task of removing old armor',
+        )
+        try {
+          await this.releaseOldReservation()
+        } catch (error) {
+          console.error('Error while performing cron task:', error)
+        }
+      },
+      {
+        scheduled: true,
+        timezone: 'Europe/Moscow',
+      },
+    )
+  }
+
+  async processPayment(userId) {
```

Comment:
Better if we move all payments to special service payments.service. In case if we have a  few types of payments